dm/checker/check_test.go

```diff
@@ -306,8 +310,14 @@ func (s *testCheckerSuite) TestShardTableSchemaChecking(c *tc.C) {
   					id int(11) DEFAULT NULL,
   					c int(11) DEFAULT NULL
 					) ENGINE=InnoDB DEFAULT CHARSET=latin1`
-
-	mock := initMockDB(c)
+	errNoSuchTable := &gmysql.MySQLError{Number: mysql.ErrNoSuchTable}
```

Comment:
is there a test to cover checkpoint existing?

Reply:
it isn't. These code mock 

```go
		checkpointSQLs := []string{
			fmt.Sprintf("SHOW CREATE TABLE %s", dbutil.TableName(instance.cfg.MetaSchema, cputil.LoaderCheckpoint(instance.cfg.Name))),
			fmt.Sprintf("SHOW CREATE TABLE %s", dbutil.TableName(instance.cfg.MetaSchema, cputil.LightningCheckpoint(instance.cfg.Name))),
			fmt.Sprintf("SHOW CREATE TABLE %s", dbutil.TableName(instance.cfg.MetaSchema, cputil.SyncerCheckpoint(instance.cfg.Name))),
		}
		var existCheckpoint bool
		for _, sql := range checkpointSQLs {
			c.tctx.Logger.Info("exec query", zap.String("sql", sql))
			_, err := instance.targetDB.DB.QueryContext(c.tctx.Ctx, sql)
			if err != nil && !utils.IsMySQLError(err, mysql.ErrNoSuchTable) {
				return err
			}
			if err == nil {
				existCheckpoint = true
				c.tctx.Logger.Info("exist checkpoint, so don't check sharding tables")
				break
			}
		}
```